questionpy/_ui.py

```diff
@@ -1,44 +1,79 @@
-        # The package has no "templates" directory, which would cause PackageLoader to raise an unhelpful ValueError.
+class _TraversableTemplateLoader(jinja2.BaseLoader):
+    """In contrast to the :class:`jinja2.FileSystemLoader` this does not support the auto-reload feature of jinja2."""
+
+    def __init__(self, traversable: Traversable):
+        self.traversable = traversable
+
+    def get_source(self, environment: jinja2.Environment, template: str) -> tuple[str, str, Callable[[], bool] | None]:
+        source_path = self.traversable.joinpath(template)
+        try:
+            source = source_path.read_text("utf-8")
+        except FileNotFoundError as e:
+            raise jinja2.TemplateNotFound(template) from e
+        else:
+            return source, template, None
+
+
+def _get_loader(package: Package) -> jinja2.BaseLoader | None:
+    templates_folder = package.get_path(f"{TEMPLATES_DIR}/")
+    if not templates_folder.is_dir():
         return None
 
-    # TODO: This looks for templates in python/<namespace>/<short_name>/templates, we might want to support a different
-    #  directory, such as resources/templates.
-    return jinja2.PackageLoader(pkg_name)
+    if os.path.exists(str(templates_folder)):
+        return jinja2.FileSystemLoader(str(templates_folder))
+    return _TraversableTemplateLoader(templates_folder)
 
 
 def create_jinja2_environment(attempt: "Attempt", question: "Question") -> jinja2.Environment:
     """Creates a Jinja2 environment with sensible default configuration.
 
     - Library templates are accessible under the prefix ``qpy/``.
-    - Package templates are accessible under the prefix ``<namespace>.<short_name>/``.
+    - Package templates are accessible under the prefix ``@<namespace>.<short_name>/``.
+    - The prefix is optional when accessing templates of the current package.
     - The QPy environment, attempt, question and question type are available as globals.
     """
     qpy_env = get_qpy_environment()
 
     loader_mapping = {}
     for package in qpy_env.packages.values():
-        loader = _loader_for_package(package)
+        loader = _get_loader(package)
         if loader:
-            loader_mapping[f"{package.manifest.namespace}.{package.manifest.short_name}"] = loader
+            loader_mapping[f"@{package.manifest.namespace}.{package.manifest.short_name}"] = loader
 
     # Add a place for SDK-Templates, such as the one used by ComposedAttempt etc.
     loader_mapping["qpy"] = jinja2.PackageLoader(__package__)
 
-    env = jinja2.Environment(autoescape=True, loader=jinja2.PrefixLoader(mapping=loader_mapping))
+    loaders: list[jinja2.BaseLoader] = [jinja2.PrefixLoader(mapping=loader_mapping)]
+
+    # Get caller package template loader.
+    module_parts = attempt.__module__.split(".", maxsplit=2)
+    if len(module_parts) < 2:  # noqa: PLR2004
+        msg = "Please do not modify the '__module__' attribute."
+        raise ValueError(msg)
```

Comment:
Vielleicht könntest du noch mit einem RegEx prüfen, ob `namespace` und `short_name` gültige Werte haben?

Das könnte vielleicht auch im Konstruktor von `PackageNamespaceAndShortName` passieren.

Reply:
Ich denke es wäre am einfachsten direkt zu schauen, ob ein Paket mit dem `namespace`/`short_name` in `qpy_env.packages` zu finden ist.

Reply:
Dan müsste allerdings in Zukunft, wenn es ein Paket-Repository gibt, immer das Repository angefragt werden statt ungültige Werte einfach abzufangen.

Reply:
Das verstehe ich nicht so ganz. `qpy_env.packages` ist ein `dict`, welches alle vom Worker geladenen Pakete beinhaltet.

Reply:
Aber das wird doch nicht immer so bleiben, oder?

Ist mir aber auch nicht wichtig.